src/Symfony/Component/PropertyInfo/PropertyInfoExtractor.php

```diff
@@ -18,33 +18,18 @@
  */
 class PropertyInfoExtractor implements PropertyInfoExtractorInterface
 {
-    /**
-     * @var PropertyListExtractorInterface[]
-     */
     private $listExtractors;
-
-    /**
-     * @var PropertyTypeExtractorInterface[]
-     */
     private $typeExtractors;
-
-    /**
-     * @var PropertyDescriptionExtractorInterface[]
-     */
     private $descriptionExtractors;
-
-    /**
-     * @var PropertyAccessExtractorInterface[]
-     */
     private $accessExtractors;
 
     /**
-     * @param PropertyListExtractorInterface[]        $listExtractors
-     * @param PropertyTypeExtractorInterface[]        $typeExtractors
-     * @param PropertyDescriptionExtractorInterface[] $descriptionExtractors
-     * @param PropertyAccessExtractorInterface[]      $accessExtractors
+     * @param iterable|PropertyListExtractorInterface[]        $listExtractors
```

Comment:
Nor is [`@final`](https://github.com/phpDocumentor/fig-standards/issues/139), its support will come when it will be adopted so imo it's not a problem.

About the redundancy I agree but we would lose all completion until editors add generics support, I don't think it's acceptable.